packages/bigcommerce-payments-integration/src/bigcommerce-payments-alternative-methods/bigcommerce-payments-alternative-methods-payment-strategy.ts

```diff
@@ -0,0 +1,325 @@
+import { noop } from 'lodash';
+
+import {
+    PayPalApmSdk,
+    PayPalSdkHelper,
+} from '@bigcommerce/checkout-sdk/bigcommerce-payments-utils';
+import {
+    InvalidArgumentError,
+    OrderFinalizationNotRequiredError,
+    OrderRequestBody,
+    PaymentArgumentInvalidError,
+    PaymentInitializeOptions,
+    PaymentIntegrationService,
+    PaymentMethodClientUnavailableError,
+    PaymentMethodInvalidError,
+    PaymentRequestOptions,
+    PaymentStrategy,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+import { LoadingIndicator } from '@bigcommerce/checkout-sdk/ui';
+
+import BigCommercePaymentsIntegrationService from '../bigcommerce-payments-integration-service';
+import {
+    ApproveCallbackPayload,
+    BigCommercePaymentsButtons,
+    BigCommercePaymentsButtonsOptions,
+    BigCommercePaymentsInitializationData,
+    NonInstantAlternativePaymentMethods,
+} from '../bigcommerce-payments-types';
+
+import BigCommercePaymentsAlternativeMethodsPaymentInitializeOptions, {
+    WithBigCommercePaymentsAlternativeMethodsPaymentInitializeOptions,
+} from './bigcommerce-payments-alternative-methods-payment-initialize-options';
+
+export default class BigCommercePaymentsAlternativeMethodsPaymentStrategy
+    implements PaymentStrategy
+{
+    private loadingIndicatorContainer?: string;
+    private orderId?: string;
+    private bigCommercePaymentsButton?: BigCommercePaymentsButtons;
+    private paypalApms?: PayPalApmSdk;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private bigCommercePaymentsIntegrationService: BigCommercePaymentsIntegrationService,
+        private bigCommercePaymentsSdkHelper: PayPalSdkHelper,
+        private loadingIndicator: LoadingIndicator,
+    ) {}
+
+    async initialize(
+        options: PaymentInitializeOptions &
+            WithBigCommercePaymentsAlternativeMethodsPaymentInitializeOptions,
+    ): Promise<void> {
+        const {
+            gatewayId,
+            methodId,
+            bigcommerce_payments_paypal, // FIXME: this option is deprecated
+            bigcommerce_payments_apms,
+        } = options;
+        const paypalOptions = bigcommerce_payments_apms || bigcommerce_payments_paypal;
```

Comment:
`bigcommerce_payments_apms` could be used instead of `paypalOptions` in current file